### PR TITLE
Fix drag + drop operations on collections pages

### LINKED ISSUE #30614: fix drag + drop interactions in collections
Adding `react-dropzone` to the collections page broke all other drag + drop interactions on that page. Specifically, the dropzone seems to be swallowing all `drop` events when `noDragEventsBubbling` is set to true. See [docs](https://react-dropzone.js.org/#section-event-propagation). Unfortunately, when that prop is false, the dropzone never receives any drop events. 😢   

Need to look at whether we can further customize drop event handling with `react-dropzone` or if we need to dump the library and write our own custom hook to more finely control event propagation.

### PR DESCRIPTION
**Pull request by iethree**
Closes https://github.com/metabase/metabase/issues/30614

### Description

Basically, `react-dropzone` was either swallowing all drop events on the collections page, or our other libraries were swallowing them all. This composes the event handlers from `react-dropzone` to only handle file drop events and allows all our drag + drop operations to play nicely together.

ALLTHEDRAGS

### How to verify

Testing:
- enable uploads in admin settings
- visit any collection
- see that you can upload files by dragging them into the main collection item list area 🥳 
- see that you can drag items around in pinned items as well 🥳 

### Checklist

The event handler code is unit tested, but I think it might be impossible to e2e test this drag behavior with cypress currently. I'll ping nemanja to see if he has any ideas.

- [x] Tests have been added/updated to cover changes in this PR

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/30718)

### CODE REVIEW

**Review by alxnddr — CHANGES_REQUESTED** [review]
It would be nice to avoid patching packages because potentially we can be stuck with upgrading them in the future.

I wonder if it is possible to find another reasonable workaround. For instance, `getRootProps()` from `useDropzone` hook returns event handlers. Can we define our own dnd event handlers that will check if the event has files and only in this case call ones from `getRootProps`?

**Comment by iethree** [comment]
> It would be nice to avoid patching packages because potentially we can be stuck with upgrading them in the future.
> 
> I wonder if it is possible to find another reasonable workaround. For instance, `getRootProps()` from `useDropzone` hook returns event handlers. Can we define our own dnd event handlers that will check if the event has files and only in this case call ones from `getRootProps`?

🤦  thanks @alxnddr , that's a much simpler solution

**Review by alxnddr — APPROVED** [review]
🚀

File: e2e/test/scenarios/collections/collection-pinned-overview.cy.spec.js [hunk 1]
```diff
@@ -199,6 +199,28 @@ describe("scenarios > collection pinned items overview", () => {
...
+
+    cy.findByTestId("pinned-items").as("pinnedItems");
+
+    // this test can give us some degree of confidence, but its effectiveness is limited
+    // because we are manually firing events on the correct elements. It doesn't seem that there's
+    // a way to actually simulate the raw user interaction of dragging a certain distance in cypress.
+    // this will not guarantee that the drag and drop functionality will work in the real world, e.g
+    // when our various drag + drop libraries start interfering with events on one another.
+    // for example, this test would not have caught https://github.com/metabase/metabase/issues/30614
+    dragAndDrop("draggingViz", "pinnedItems");
```
Comment (iethree) [inline e2e/test/scenarios/collections/collection-pinned-overview.cy.spec.js:222]: @nemanjaglumac I wonder if you know if there's any way to test drag + drop that doesn't involve specifying which events to fire on which elements? that actually doesn't help catch bugs that are caused by events not getting fired on the intended elements when they're dragged over.

Comment (iethree) [reply e2e/test/scenarios/collections/collection-pinned-overview.cy.spec.js]: I took a close look at both https://github.com/dmtrKovalenko/cypress-real-events and https://github.com/4teamwork/cypress-drag-drop and neither allow you to do what we really need here - simulate a mouse moving over a screen region - they both rely on firing events on specific elements - the bugfix here is that the events got fired on the wrong elements while hovering the right screen region - so any test where we tell it where the fire the events will not fail. 😢

**Pull request merged**